packages/devextreme/js/__internal/ui/chat/messagelist.ts

```diff
@@ -169,6 +176,48 @@ class MessageList extends Widget<Properties> {
     });
   }
 
+  _shouldAddDayHeader(timestamp: undefined | string | number | Date): boolean {
+    const { showDayHeaders } = this.option();
+
+    if (timestamp === undefined || !showDayHeaders) {
+      return false;
+    }
+
+    const deserializedDate = dateSerialization.deserializeDate(timestamp);
+
+    if (!isDate(deserializedDate) || isNaN(deserializedDate.getTime())) {
+      return false;
+    }
+
+    return !dateUtils.sameDate(this._lastMessageDate, deserializedDate);
+  }
+
+  _createDayHeader(timestamp: string | number | Date | undefined): void {
+    const deserializedDate = dateSerialization.deserializeDate(timestamp);
+    const today = new Date();
+    const yesterday = new Date(new Date().setDate(today.getDate() - 1));
+    this._lastMessageDate = deserializedDate;
+
+    let headerDate = deserializedDate.toLocaleDateString(undefined, {
+      day: '2-digit',
+      month: '2-digit',
+      year: 'numeric',
+    }).replace(/[/-]/g, '.');
+
+    if (dateUtils.sameDate(deserializedDate, today)) {
+      headerDate = `${messageLocalization.format('Today')} ${headerDate}`;
+    }
+
+    if (dateUtils.sameDate(deserializedDate, yesterday)) {
+      headerDate = `${messageLocalization.format('Yesterday')} ${headerDate}`;
```

Comment:
Did I understand correctly that the format 'Today' + date() comes from the designer? I haven't explored other products and will leave this to your discretion.

Reply:
It came from the card i'm working on
<img width="551" alt="image" src="https://github.com/user-attachments/assets/99dc67e7-a0a2-4794-ac84-638a5cf460ae">